deepchem/models/torch_models/ferminet.py

```diff
@@ -0,0 +1,336 @@
+"""
+Implementation of the Ferminet class in pytorch
+"""
+
+try:
+  import torch
+  from torch import Tensor
+  import torch.nn as nn
+  import torch.nn.functional as F
+except ModuleNotFoundError:
+  raise ImportError('These classes require PyTorch to be installed.')
+
+from typing import List, Optional, Any, Tuple
+from rdkit import Chem
+import numpy as np
+from deepchem.utils.molecule_feature_utils import ALLEN_ELECTRONEGATIVTY
+from deepchem.utils.geometry_utils import compute_pairwise_distances
+
+from deepchem.models.torch_models import TorchModel
+import deepchem.models.optimizers as optim
+from deepchem.utils.electron_sampler import ElectronSampler
+
+
+def test_f(x: np.ndarray) -> np.ndarray:
+  # dummy function which can be passed as the parameter f. f gives the log probability
+  # TODO replace this function with forward pass of the model in future
+  return 2 * np.log(np.random.uniform(low=0, high=1.0, size=np.shape(x)[0]))
+
+
+class FerminetModel(torch.nn.Module):
```

Comment:
Doing a drive by review, so apologies if you've already discussed this. The usual convention we have is that classes ending in `Model` should inherit from `TorchModel`. So `FerminetModel` should inherit from `TorchModel`. Do you have a `TorchModel` wrapper in place yet?

Reply:
Thanks for the comment, I should interchange the 2 classes name here - Ferminet and FerminetModel